examples/visualization/plot_study.py

```diff
@@ -0,0 +1,92 @@
+"""
+Optuna example that visualizes the optimization result of multi-layer perceptrons.
+
+In this example, we optimize the validation accuracy of object recognition using
+scikit-learn and Fashion-MNIST. We optimize a neural network. As it is too time
+consuming to use the whole Fashion-MNIST dataset, we here use a small subset of it.
+
+We have can execute this example as follows.
+
+    $ python plot_study.py
+
+**Note:** If a parameter contains missing values, a trial with missing values is not plotted.
+"""
+
+
+from sklearn.datasets import fetch_openml
+from sklearn.model_selection import train_test_split
+from sklearn.neural_network import MLPClassifier
+
+import optuna
+from optuna.visualization import plot_contour
+from optuna.visualization import plot_intermediate_values
+from optuna.visualization import plot_optimization_history
+from optuna.visualization import plot_parallel_coordinate
+from optuna.visualization import plot_slice
+
+
+def build_objective():
+    fmnist = fetch_openml(name="Fashion-MNIST", version=1)
+    classes = list(set(fmnist.target))
+
+    # For demonstrational purpose, only use a subset of the dataset.
+    n_samples = 4000
+    data = fmnist.data[:n_samples]
+    target = fmnist.target[:n_samples]
+
+    x_train, x_test, y_train, y_test = train_test_split(data, target)
+
+    def objective(trial):
+
+        clf = MLPClassifier(
+            hidden_layer_sizes=tuple(
+                [trial.suggest_int("n_units_l{}".format(i), 32, 64) for i in range(3)]
```

Comment:
[nit] We can also optimize the number of layers, but I'd also not optimize it in this example since the objective is to make the visualization feature in use.